harness/tests/cli/test_experiment.py

```diff
@@ -92,14 +94,14 @@ def test_wait_returns_error_code_when_experiment_errors(
 ) -> None:
     auth_mock.return_value = mock_det_auth()
 
-    exp = api_server.sample_get_experiment().experiment
+    with open(pathlib.Path(__file__).parent.parent / "fixtures" / "experiment_flaky.json") as f:
```

Comment:
I'd had in mind hacking in the ability to force an attribute after fixtures were loaded (and maintaining a single fixture), but this is fine.

Mmm... another proposal: don't worry about broken tests once we've decided the breakage is due to the test's brittleness itself and not a bad decision in the tested code? I kind of wish we'd discussed this before you spent time on the fix, and I understand if you don't want to throw that work away.

Reply:
yeah, i attempted to do this at first. the issue is passing in attributes (like ID) to `api_server`. it's not hard persay, just requires some thought. like, should the call to `api_server` accept a custom config somehow? it seems a bit too general for that, so then maybe it should accept a mapping of custom request path -> functions? the problem with that is that some functions need to subclass `SimpleHTTPRequestHandler`. 

i figured i'd just get it to work for now, and once you have a better idea of the SDK test refactoring, then we should definitely refactor this.